### PR TITLE
POPR-164: Added add slide button to editor view component

### PR DESCRIPTION
**Pull request by maruPortal**
# Related Task
- https://app.clickup.com/t/459857/POPR-164

# Proposed Changes
- created generalIconButton on UI
- Added button on editor view component
- When clicking on that button a slide is added to an array of slides

# Additional Info (Optional)
- 

# How to test this?
- click on the add slide button

# Task Checklist
- [x] All new features have been documented
- [x] Conventions and Acceptance criteria have been met
- [x] Code is free of warnings
- [x] ClickUp task has been set to Awaiting Review

### CODE REVIEW

File: frontend/app/src/layouts/PreviewLayout/index.tsx [hunk 1]
```diff
@@ -1,22 +1,34 @@
...
-  const { t } = useTranslation(['config', 'translations']);
-  const { data: contacts } = useQuery(['getContacts'], async () => await getContacts('63120fd98028831ecf43b444'));
+  const exampleSlide = { header: 'header', main: 'some text', footer: 'footer' };
+  const [slides, setSlides] = useState([exampleSlide]);
+
+  const addSlide = () => {
+    setSlides([...slides, exampleSlide]);
+  };
+
+  const showSlides = () => {
```
Comment (rosariosm) [inline frontend/app/src/layouts/PreviewLayout/index.tsx]: Move this to a  component called SlideList. You can define it in app project.  Move all Slide logic to that componenr

File: frontend/ui/src/components/GeneralIconButton/styles.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  padding: 16px;
+  gap: 8px;
+
+  width: 3.5rem;
+  height: 3.5rem;
+
+  background: #451d53;
+  color: white;
+
+  box-shadow: 0px 3px 5px -1px rgba(0, 0, 0, 0.2),
```
Comment (rosariosm) [inline frontend/ui/src/components/GeneralIconButton/styles.ts:8]: I don't think you need to redefine all this styles. All you need to do is use IconButton size='large' and set background color from theme :)

File: frontend/ui/src/components/GeneralIconButton/styles.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    0px 6px 10px rgba(0, 0, 0, 0.14), 0px 1px 18px rgba(0, 0, 0, 0.12);
+  border-radius: 28px;
+
+  flex: none;
+  order: 0;
+  flex-grow: 0;
+
+  &:hover {
```
Comment (rosariosm) [inline frontend/ui/src/components/GeneralIconButton/styles.ts]: Would you mind adding a Material ui theme and edit colors there? I can help you if you need it

**Review by rosariosm — CHANGES_REQUESTED** [review]

File: frontend/ui/src/components/GeneralIconButton/styles.ts (hunk 3 above)
Comment (rosariosm) [reply frontend/ui/src/components/GeneralIconButton/styles.ts]: Talk to Pam because I think she was implementing it, but I'm not sure

**Review by AgusPk — APPROVED** [review]

**Pull request merged**